lib/middleware/add-weather.js

```diff
@@ -0,0 +1,24 @@
+const getLocation = require('../services/maps-api');
+const getForecast = require('../services/weather-api');
+
+module.exports = () => (req, res, next) => {
+  const { address } = req.body;
```

Comment:
Instead of calling getLocation inside of getForecast, you can get the location from your req.body in your getForecast function. getLocation adds the location to the req.body. So at the top here, you should have ```const { location } = req.body```.